cmd/thv-operator/test-integration/mcp-server/mcpserver_runconfig_integration_test.go

```diff
@@ -547,5 +547,115 @@ var _ = Describe("RunConfig ConfigMap Integration Tests", func() {
 			Expect(runConfig.CmdArgs).To(Equal([]string{"--arg1", "--arg2", "--arg3"}))
 			Expect(runConfig.ToolsFilter).To(Equal([]string{"tool3", "tool1", "tool2"}))
 		})
+
+		It("Should handle MCPServer with OIDC authentication configuration", func() {
+			namespace := "oidc-test-ns"
+			mcpServerName := "oidc-server"
+			configMapName := mcpServerName + "-runconfig"
+
+			// Create namespace
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
+			_ = k8sClient.Create(ctx, ns)
+
+			// Create MCPServer with OIDC configuration
+			mcpServer := &mcpv1alpha1.MCPServer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      mcpServerName,
+					Namespace: namespace,
+				},
+				Spec: mcpv1alpha1.MCPServerSpec{
+					Image:     "auth/mcp-server:latest",
+					Transport: "stdio",
+					ProxyPort: 8080,
+					OIDCConfig: &mcpv1alpha1.OIDCConfigRef{
+						Type: "inline",
+						Inline: &mcpv1alpha1.InlineOIDCConfig{
+							Issuer:             "https://auth.example.com",
+							Audience:           "toolhive-api",
+							JWKSURL:            "https://auth.example.com/.well-known/jwks.json",
+							IntrospectionURL:   "https://auth.example.com/oauth/introspect",
+							ClientID:           "toolhive-client",
+							ClientSecret:       "secret123",
+							JWKSAllowPrivateIP: true,
+						},
+					},
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, mcpServer)).Should(Succeed())
+			defer k8sClient.Delete(ctx, mcpServer)
```

Comment:
nit: Gingko has construct to run after a context so you wouldn't necessarily need to do "defer".